tests/cases/conformance/externalModules/typeValueMerge1.ts

```diff
@@ -0,0 +1,17 @@
+// @noEmit: true
+
+// @Filename: other.ts
+export type A = string;
+function A() {}
+export { A };
```

Comment:
this resolves to a value in this situation so it's kinda equivalent to:
```ts
export type A = string;
export function A() {}
```
and this was always~ allowed.

Reply:
Note that in the past this was considered to be ambiguous and one could get an error when writing code like this ("Individual declarations in merged declaration 'B' must be all exported or all local"). In the referenced issue it was observed that the variant with exported function declaration **didn't** report an error (at least back in 2016 :P). @DanielRosenwasser even stated that it should error. However, since the behavior was changed in 4.9 by https://github.com/microsoft/TypeScript/pull/50853 I think that the proper resolution nowadays is to accept the new behavior and just commit those tests.